### ISSUE TITLE
Swiping the month feels bad

### ISSUE THREAD

**Issue by curliq** [roomorama/Caldroid#375]
Hey all. I haven't done any change in the caldroid files, just implemented it. When I swipe to the next month it feels really bad, kind of breaks in the middle of the animation, really unsmooth, this doesnt happen in the emulator, just in my phone (1Gb RAM; 1.3Ghz Quad-core, 5'', API 21).
Does this happen due to my phone's low specs or is there something to do to optimize the caldroid?
Thanks

**Comment by thomasdao** [comment]
You are right, Caldroid is a bit heavy fragment and it probably does not run very smoothly on low end device. To support smooth swiping, it uses a `InfiniteViewPager` which contains 4 gridview fragment, and setup 4 gridview fragment takes time. Whenever user swipes to new month, it needs to create an array list of `DateTime` for new month as well. I have tried hard to reduce amount of work when user swipes so it feels smoother, but probably there's more can be done.

**Comment by chuks** [comment]
Have you been able to figure out how to achieve a smooth swipe? There's gotta be a way this can be achieve, i.e. look at the Google Calendar for Android, it's super smooth. I'm trying to achieve something similar with Caldroid i.e. there are data represented as small dots, like the Google Calendar on each date, and each updates as the month is swiped to (data is paged from local source or database, and shouldn't affect performance). Has anyone been able to achieve this?

If I preset all the data on each cell, and never refresh, it swipes smoothly. However, once I begin updating the month data according to swipes, it feels laggy.

**Comment by curliq** [comment]
@chuks try this lib https://github.com/prolificinteractive/material-calendarview works for me

**Comment by chuks** [comment]
@joaosardinha You are a life saver! Been checking that one out --works like a charm. One thing I haven't been able to achieve yet is animate the layout change when switching from month view mode to week view mode, and vice versa. The default behavior is just to jump between both modes (which is OK, but not the best UX).

**Comment by curliq** [comment]
@chuks Glad I helped. I havent used the week mode, but I'm guessing it's gonna take a litle work to put an animation on it

**Comment by chuks** [comment]
@joaosardinha yeah,  guessing too. Unless what's underneath can work with the standard animate layout change property of the ViewGroup.

**Comment by curliq** [comment]
Do you mean setting animateLayoutChanges to true in the container layout? if so, it doesn't I tried :(

**Comment by chuks** [comment]
Yeah, that's the one. I tried too, but it doesn't work.

**Comment by curliq** [comment]
Well if you're trying to go for it, good luck!
